clients/src/main/java/org/apache/kafka/common/metrics/stats/Rate.java

```diff
@@ -82,7 +82,7 @@ public long windowSize(MetricConfig config, long now) {
          * but this approach does not account for sleeps. SampledStat only creates samples whenever record is called,
          * if no record is called for a period of time that time is not accounted for in windowSize and produces incorrect results.
          */
-        long totalElapsedTimeMs = now - stat.oldest(now).lastWindowMs;
+        long totalElapsedTimeMs = now - stat.oldest(now).startTimeMs;
```

Comment:
Could we add some comments to note that the total elapse time could be larger than the window size since we keep the overlapping samples?

Reply:
Applied